lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+      }
+    }
+    else
+    {
+      $criteria->add(QubitSlug::SLUG, $uniqueValue);
+
+      if (null === $slug = QubitSlug::getOne($criteria))
+      {
+        throw new Exception("Invalid information object slug '$uniqueValue'");
+      }
+
+      if (null === $informationObject = QubitInformationObject::getById($slug->objectId))
+      {
+        throw new Exception("Missing information object for slug '$uniqueValue'");
+      }
+    }
+
+    return $informationObject;
+  }
+
+  /**
+   * Check to see if an object has already been imported and, if not, print
+   * text to notify.
+   *
+   * @param string $filepath  Path, within DIP, to file
+   *
+   * @return void
+   */
+  protected function auditDigitalObject($filepath)
+  {
+    $filename = basename($filepath);
+
+    $query = "SELECT id FROM digital_object WHERE name=?";
+
+    $statement = QubitFlatfileImport::sqlQuery($query, array($filename));
+
+    if (!$statement->fetchColumn())
+    {
+      $this->log("Missing ". $filename);
+    }
+  }
+
+  /**
+   * Import object, associate with information object, and, optionally, log
+   *
+   * @param object QubitInformationObject $informationObject  Information object
+   * @param string $filepath  Path, within DIP, to file
+   * @param boolean $undoLog  Optional undo log location
+   * @param boolean $container  Whether information object contains others
+   *
+   * @return void
+   */
+  protected function addDigitalObject($informationObject, $filepath, $undoLog = null, $container = false)
+  {
+    // Abort if a digital object already exists for this information object
+    if (null !== $informationObject->getDigitalObject())
+    {
+      $this->log("A digital object is already attached to $informationObject->identifier (slug: $informationObject->slug). Skipping.");
+
+      return;
+    }
+
+    // Make sure file exists
+    if (!file_exists($filepath))
+    {
+      throw new Exception("Couldn't find file '$filepath'");
+    }
+
+    // Create digital object
+    $this->log("Importing '$filepath'");
+
+    $do = new QubitDigitalObject;
+    $do->usageId = QubitTerm::MASTER_ID;
+    $do->assets[] = new QubitAsset($filepath);
+
+    // Add digital object to information object
+    $informationObject->digitalObjects[] = $do;
+
+    // Add DIP UUID as aipUUID information object property
+    if (null !== $dipUUID = $this->getUUID(basename($this->dipDir)))
+    {
+      $this->log('Creating property: dip UUID '. $dipUUID);
+      $informationObject->addProperty('aipUUID', $dipUUID);
+    }
+
+    // Add object UUID as objectUUID information object property
+    if (null !== $objectUUID = $this->getUUID(basename($filepath)))
+    {
+      $this->log('Creating property: object UUID '. $objectUUID);
+      $informationObject->addProperty('objectUUID', $objectUUID);
+    }
+
+    // Save and, optionally, log
+    $informationObject->save($this->conn);
+
+    if (isset($undoLog))
```

Comment:
I think it will always be null until we send it to 'importDigitalObjects()' first or we make it a global variable.